### ISSUE TITLE
anaconda-project should provide plugins cookiecutter functionality

### ISSUE THREAD

**Issue by fpliger** [Anaconda-Platform/anaconda-project#102]
After #89 lands we also need a way to help users creating plugins... I.e. some cookiecutter command that would prepare the project plugin and drive the user to customize it, package it and try it out...

**Comment by bollwyvl** [comment]
Awesome. In addition to generating a canonical plugin development project (right?), core or plugins could include a number of other cookiecutters:
- whole projects
- a command, either in an existing env_spec or a new one (with [hooks](https://cookiecutter.readthedocs.io/en/latest/advanced/hooks.html) that do project things)

While we _should_ distribute the actual cookiecutters in the end-user package, we may not want `cookiecutter` directly in the dependency chain (it carries a number of dependencies that duplicate things already in this repo... another yaml parser, etc)... but still be available, tested, and documented.

Riffing on the "private environment" idea, one approach to this would be to create a temporary project that has a (pinned) version of cookiecutter:

```
$> anaconda-project template plugin
`cookiecutter` is not installed in the same environment as `anaconda-project`. 

Use a private project? [Y/n]
```

Something like this:
```yaml
name: cookiecutter-anaconda-project-plugin-bootstrap
packages:
- cookiecutter=1.5.1
commands:
  cookiecutter:
    unix: cookiecutter
    windows: cookiecutter
```

**Comment by mcg1969** [comment]
Thanks for opening this issue, and apologies for the long delay in responding.

Anaconda Project has not been actively developed in some time and has largely been superseded by newer alternatives:

* [pixi](https://pixi.sh/) and [uv](https://docs.astral.sh/uv/) represent the first truly successful successors.
* [conda-workspaces](https://github.com/conda-incubator/conda-workspaces) is a promising pixi-compatible alternative for users who prefer to stay closer to the conda ecosystem.

Going forward, work on this repository is focused on critical bug fixes and on tooling to help users migrate existing Anaconda Project projects to pixi (and, by extension, conda-workspaces). See the [Pixi support](https://anaconda-project.readthedocs.io/en/latest/pixi-support.html) page in our documentation for details on the `anaconda-project export-pixi` command, which can convert an existing `anaconda-project.yml` into a `pixi.toml` for you.

The `anaconda-project` package itself will remain available, so existing projects relying on its stable feature set can continue to be supported. However, we do not anticipate further enhancement work, and for that reason we are closing out the bulk of our open issue backlog — including this one.

If this issue describes a critical regression in the stable feature set (as opposed to an enhancement or an edge case in a feature we no longer plan to expand) and you have a minimal reproducer against a current release, please feel free to reopen and we will take another look.

Thank you for your interest in Anaconda Project.
